test/metabase/query_processor_test/date_time_zone_functions_test.clj

```diff
@@ -132,3 +142,140 @@
                        :breakout    [[:expression "expr"]]}]]]
               (testing (format "%s function works as expected on %s column for driver %s" op col-type driver/*driver*)
                 (is (= (set expected) (set (test-date-extract query))))))))))))
+
+(defn- date-math
+  [op x amount unit col-type]
+  (let [amount (if (= op :date-add)
+                 amount
+                 (- amount))
+        result (cond
+                 ;; unlike everybody else, vertica decided when adding 1 year it will do that by adding 365 days,
+                 ;; or 90 days if you want to add a quarter.
+                 ;; That means if you do `select date '2004-02-02' + interval '2 year';` in vertica, it will returns
+                 ;; `2006-02-01` because 2004 is a leap year. All others DBs will return `2006-02-02` instead.
```

Comment:
Don't we want to eliminate this difference? (I could argue pro and contra, so I'm just wondering.)

Reply:
https://github.com/metabase/metabase/pull/25414/commits/9d32c7b56af1f55b292678d582d0e3183f7078f4

I make the behavior of vertica consistent with other dbs in this commit. WDYT?